### ISSUE TITLE
YellowBox warnings when using with React Native

### ISSUE THREAD

**Issue by kadikraman** [FormidableLabs/urql#677]
**react-native:** 0.62.0
**react-navigation:** ^5.1.1
**urql:** ^1.9.5

Using urql with React Native seems to cause a lot of yellow-box warnings when navigating.

![Screenshot 2020-04-01 at 11 03 50](https://user-images.githubusercontent.com/6534400/78148379-b6a6f700-742c-11ea-8c0c-2a12d31e90b4.png)

We have the default exchanges + an `authExchange` for adding the auth header asynchronously

```js
const client = createClient({
  url: "http://localhost:3000/graphql",
  exchanges: [dedupExchange, cacheExchange, authExchange, fetchExchange],
});
```
**Steps to reproduce**

1. Create a React Native project with `react-navigation` and at least two screens to navigate between
2. Do a `query` using `urql` with the `useQuery` hook on each of the pages
3. Navigate between the screens

**Expected behavior**
No warnings

**Actual behavior**
Navigating between the screens will produce at least one warning (`didSendNetworkData` is the most common)

**Comment by kitten** [comment]
Does this happen when in-flight requests are cancelled (which would be sporadic) or consistently for each query?

**Comment by kadikraman** [comment]
I seem to get `didSendNetworkData` consistently for each query but only after navigating. It doesn't happen when I navigate to a page that _doesn't_ have a query. It only happens when going from Page1 to Page2 and both have a query.

**Comment by kitten** [comment]
This may be related to: https://github.com/facebook/react-native/issues/12985

**Comment by kadikraman** [comment]
Found the culprit! It's was the auth exchange all along. Looks like I can reliably reproduce the error by doing an async call to Sensitive Info or Async Storage in the exchange.

I made a repo to reproduce it, specifically [here](https://github.com/kadikraman/UrqlTest/blob/master/App.js#L19) both calls to `AsyncStorage.getItem` and `SInfo.getItem` will cause the error, but just  resolving a promise with a timeout does not.

Looks like we'll have to rethink how we're adding the auth header, but this is probably not an urql issue.

Thanks for your help in debugging this! 💯
